WcaOnRails/app/models/regulation.rb

```diff
@@ -0,0 +1,16 @@
+class Regulation < SimpleDelegator
+  begin
+    REGULATIONS = JSON.parse(File.read(Rails.root.to_s + "/app/views/search_results/wca-regulations.json")).freeze
+  rescue
+    REGULATIONS = [].freeze
+  end
+
+  def limit(number)
+    first(number)
+  end
+
+  def self.search(query, *)
+    query.downcase!
+    Regulation.new(REGULATIONS.select { |reg| (reg['text'].downcase.include?(query) || reg['id'].downcase.include?(query)) })
```

Comment:
Thanks, close enough!
It was just missing a `.dup` to do more than a reference copy on `REGULATIONS` and the `.each` on the split :)